bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubBridgeHandler.java

```diff
@@ -0,0 +1,155 @@
+    private final Bundle bundle;
+
+    private SiteApi siteApi;
+    private String bridgeId = "";
+
+    public MetOfficeDataHubBridgeHandler(final Bridge bridge, @Reference HttpClientFactory httpClientFactory,
+            @Reference TranslationProvider translationProvider, @Reference LocaleProvider localeProvider,
+            @Reference StorageService storageService, @Reference TimeZoneProvider timeZoneProvider) {
+        super(bridge);
+        bridgeId = getThing().getUID().getAsString();
+        this.translationProvider = translationProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.siteApi = new SiteApi(bridgeId, httpClientFactory, storageService, translationProvider, localeProvider,
+                timeZoneProvider, scheduler);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        siteApi.registerListeners(bridgeId, this);
+
+        config = getConfigAs(MetOfficeDataHubBridgeConfiguration.class);
+
+        siteApi.setLimits(config.siteRateDailyLimit);
+        siteApi.setApiKey(config.siteApiKey);
+        siteApi.validateSiteApi();
+    }
+
+    @Override
+    public void dispose() {
+        siteApi.deregisterListeners(bridgeId, this);
+        siteApi.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    // API Management
+
+    public SiteApi getSiteApi() {
+        return siteApi;
+    }
+
+    // Localization functionality
+
+    public String getLocalizedText(String key, @Nullable Object @Nullable... arguments) {
+        String result = translationProvider.getText(bundle, key, key, localeProvider.getLocale(), arguments);
+        return Objects.nonNull(result) ? result : key;
+    }
+
+    // Implementation of IRateLimiterListener
+
+    @Override
+    public void processRateLimiterUpdated(RequestLimiter requestLimiter) {
+        final Map<String, String> newProps = new HashMap<>();
+        newProps.put(BRIDGE_PROP_FORECAST_REQUEST_COUNT, String.valueOf(requestLimiter.getCurrentRequestCount()));
+        this.updateProperties(newProps);
+    }
+
+    // Implementation of IConnectionStatusListener
+
+    @Override
+    public void processAuthenticationResult(boolean authenticated) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                getLocalizedText("bridge.error.site-specific.auth-issue"));
+    }
+
+    @Override
+    public void processCommunicationFailure(final @Nullable Throwable e) {
+        String message = "";
+        if (e != null) {
+            if (e.getLocalizedMessage() != null) {
+                message = e.getLocalizedMessage();
+            } else if (e.getMessage() != null) {
+                message = e.getMessage();
+            }
+        }
+        if (message == null || message.isBlank()) {
+            message = "?";
```

Comment:
maybe change to `unknown` as that is more clear then a question mark

Reply:
sure I have updated to a i18n text so it can be modified if required.